### DISCUSSION TITLE
Could we get a global unhandled exception handler?

### DISCUSSION THREAD

**Discussion by amp64** [AvaloniaUI/Avalonia#17650] [Q&A]
### Is your feature request related to a problem? Please describe.

I need to generate telemetry when my app has an unhandled exception. I have manually added handlers to Main for Windows and iOS, but can't figure out where to put one for Android.

### Describe the solution you'd like

Would be nice if Application had an override that would be called when an unhandled exception occurred so it could be set up in the portable code layer.

**Comment by maxkatz6** [comment]
There is one: `Dispatcher.UIThread.UnhandledException`. Implementation follows WPF rules, if you familiar with that one.
And it only can intercept exceptions from Avalonia dispatcher operations, that couldn't have been handled in other way.

**Reply by maxkatz6** [reply]
For anything else, you can wait for https://github.com/dotnet/runtime/issues/101560 being implemented on runtime level
